exercises/flatten-array/src/example.h

```diff
@@ -0,0 +1,19 @@
+#ifndef FLATTEN_ARRAY_H
+#define FLATTEN_ARRAY_H
+
+#define ARRAY_SIZE(arr) (sizeof(arr)/sizeof(arr[0]))
+#define NONE 0
+
+enum item_type { UNDEF, null, ELEMENT, ARRAY };
+struct item {
+   int value;
+   struct item *next;
+   struct item *array_elements;
+   enum item_type type;
+};
+
+typedef struct item item_t;
```

Comment:
If you put this before the struct definition then you can use `item_t *` in the struct instead of `struct item *`